apps/web/server/auth.ts

```diff
@@ -77,17 +88,34 @@ const providers: Provider[] = [
       email: { label: "Email", type: "email", placeholder: "Email" },
       password: { label: "Password", type: "password" },
     },
-    async authorize(credentials) {
+    async authorize(credentials, req) {
       if (!credentials) {
+        logAuthenticationError(
+          "<unknown>",
+          "Credentials missing",
+          req.headers?.["x-forwarded-for"],
+        );
         return null;
       }
 
       try {
-        return await validatePassword(
+        const isValidPassword = await validatePassword(
           credentials?.email,
           credentials?.password,
         );
+        logAuthenticationError(
+          credentials?.email,
+          "Password invalid",
+          req.headers?.["x-forwarded-for"],
+        );
+        return isValidPassword;
       } catch (e) {
+        const error = e as Error;
+        logAuthenticationError(
+          credentials?.email,
+          error.message,
+          req.headers?.["x-forwarded-for"],
```

Comment:
Let's use (https://github.com/pbojinov/request-ip) to get the real IP. It's much more complicated than just the `x-forwarded-for`.